src/libs/highlevel/README.md

```diff
@@ -0,0 +1,300 @@
+* **(Unsigned) Long Long**: a 64-bit (unsigned) integer, represented by `KDB_TYPE_LONG_LONG` (`KDB_TYPE_UNSIGNED_LONG_LONG`) in metadata
+* **Float**: whatever your compiler treats as `float`, probably IEEE-754 single-precision, represented by `KDB_TYPE_FLOAT` in metadata
+* **Double**: whatever your compiler treats as `double`, probably IEEE-754 double-precision, represented by `KDB_TYPE_DOUBLE` in metadata
+* **Long Double**: whatever your compiler treats as `long double`, not always available, represented by `KDB_TYPE_LONG_DOUBLE` in metadata
+
+## Setup
+
+First you have to add `elektra-highlevel` to the linked libraries of your application. To be able to use it in your source file,
+just include the main header with `#include <elektra.h>` at the top of your file.
+
+## Core Concepts
+
+### Struct `Elektra`
+`Elektra` is the handle you use to access the underlying KDB (hierarchical key database) that stores the configuration key-value pairs.
+All key-value read and write operations expect this handle to be passed as in as a parameter. To create the handle, you simply write:
+
+```c
+ElektraError * error = NULL;
+Elektra * elektra = elektraOpen ("/sw/org/myapp/#0/current", NULL, &error);
+```
+
+Please replace `"/sw/org/myapp/#0/current"` with an appropriate value for your application (see [Namespaces](/doc/tutorials/namespaces.md)
+for more information). You can use the parameter `defaults` to pass a KDB `KeySet` containing `Key`s with default values to the `Elektra`
+instance.
+
+The passed in `ElektraError` can be used to check for initialization errors. You can detect initialization errors by comparing the result of
+`elektraOpen` to NULL:
+
+```c
+if (elektra == NULL)
+{
+  // handle the error, e.g. print description
+  elektraErrorReset(&error);
+}
+```
+
+If an error occurred, you MUST call `elektraErrorReset` before using the same error pointer in any other function calls (e.g. elektraSet*
+calls). It is also safe to call `elektraErrorReset`, if no error occurred.
+
+
+In order to give Elektra the chance to clean up all its allocated resources, you have to close your instance, when you are done using it,
+by calling:
+
+```c
+elektraClose (elektra);
+```
+
+*NOTE:* Elektra is only thread-safe when you use one handle per thread or protect your handle. If you have multiple threads accessing
+key-values, create a separate handle for each thread to avoid concurrency issues.
+
+#### Configuration
+Currently there is only one way to configure an `Elektra` instance:
+
+```c
+void elektraFatalErrorHandler (Elektra * elektra, ElektraErrorHandler fatalErrorHandler)
+```
+This allows you to set the callback called by Elektra, when a fatal error occurs. Technically a fatal error could occur any time, but
+the most common use case for this callback is inside of functions that do not take a separate `ElektraError` argument. For example,
```

Comment:
We should also consider to use the fatal error handler if the user passed 0 to the functions that take an ElektraError argument.